src/modules/job-exec/job-exec.c

```diff
@@ -1061,9 +1062,11 @@ int mod_main (flux_t *h, int argc, char **argv)
 
     rc = flux_reactor_run (flux_get_reactor (h), 0);
 out:
+    saved_errno = errno;
     if (flux_event_unsubscribe (h, "job-exception") < 0)
         flux_log_error (h, "flux_event_unsubscribe ('job-exception')");
     job_exec_ctx_destroy (ctx);
+    errno = saved_errno;
```

Comment:
Ug, we should fix that in `flux_event_unsubscribe()`.  I'll open an issue on it.